providers/implementations/kdfs/pbkdf2.c

```diff
@@ -330,20 +401,18 @@ static int pbkdf2_derive(const char *pass, size_t passlen,
         return 0;
     }
 
+#ifndef FIPS_MODULE
```

Comment:
This looks wrong.  Why is this code only being included outside the FIPS boundary?
Is it even right to have these checks outside of FIPS?

Reply:
For FIPS provider, these checks are performed in the `fips_lower_bound_check_passed` function, which is called in the `kdf_pbkdf2_derive` function.

Reply:
Why are we doing these checks outside of the FIPS provider?
I thought they were a NIST imposed restriction.

Reply:
These checks existed in the previous version (< 3.4.0).

To avoid the behavior change, I just keep the checks for default provider here, and perform checks for FIPS provider in the `kdf_pbkdf2_derive` function.

However, if the behavior change is acceptable, I can remove this block so that only FIPS provider performing these checks.

Reply:
The checks are security hardening, let's keep them.